components/PageComponents/index.tsx

```diff
@@ -0,0 +1,35 @@
+import CallToActionButton from "@/components/CallToActionButton";
+import Hero from "@/components/Hero";
+import Component from "@/types/Component";
+import NoComponentsMessage from "../NoComponentsMessage";
+
+// Mapping Sanity components to React components
+const componentMap: Record<Component["_type"], React.ComponentType<any>> = {
+  hero: Hero,
+  callToAction: CallToActionButton,
+};
+
+type PageComponentsProps = {
+  components: Component[];
+};
+
+export default function PageComponents({ components }: PageComponentsProps) {
+  return (
+    <main>
+      {components && components.length > 0 ? (
```

Comment:
If we're checking for an empty state, let's instead pull this outside of the return to simplify the code

```
if(!components?.length) return <NoComponentsMessage />

return (
  <main>
    ...
  </main>
)
```

Reply:
I moved this logic up in 9314e77